File: src/main/java/com/zw/cf/service/ProductService.java
package com.zw.cf.service;

import com.zw.cf.model.Product;
import com.zw.cf.vo.ProductListFind;
import com.zw.plug.Response;

import java.util.List;

/**
 * Created by zhaowei on 2017/8/17.
 */
public interface ProductService {
    Response add(Product product);
    Response addList(List<Product> list);
    Response list(Integer pageNum, Integer pageSize, ProductListFind productListFind);
    Response getById(String id);
    Response update(Product product);
    Response del(String id);
}

File: frontend/src/app/common/restService/ProductService.ts
/**
 * Created by zhaowei on 2017/10/16.
 */
import {Injectable} from '@angular/core';
import {HttpServer} from '../service/HttpServer';
@Injectable()
export class ProductService {

  url = '/cfmy/product/:params1/:params2/:params3/:params4/:params5';
  urls = {
    add: {
      method: 'post',
      params: {
        params1: 'add'
      }
    },
    addList: {
      method: 'post',
      params: {
        params1: 'addList'
      }
    },
    list: {
      method: 'post',
      params: {
        params1: 'list'
      }
    },
    getById: {
      method: 'get',
      params: {
        params1: 'getById'
      }
    },
    update: {
      method: 'post',
      params: {
        params1: 'update'
      }
    },
    del: {
      method: 'get',
      params: {
        params1: 'del'
      }
    }
  };
  constructor( httpServer: HttpServer) {
    httpServer.S(this);
  }
}
